### PR TITLE
commit 2 update

### PR DESCRIPTION
**Pull request by nipunupulitha**

### CODE REVIEW

**Comment by nipunupulitha** [comment]
update

File: app/Http/Controllers/PostController.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        $request->validate([
+
+            // 'user_id' => auth()->user()->id,
+
+            'title' => 'required',
+            'description' => 'required',
+
+
+        ]);
+        $postRepo = new Post();
```
Comment (ccdilan) [inline app/Http/Controllers/PostController.php:58]: what is the purpose of creating new Post object here. this part also done in repository?

File: tests/Unit/PostTest.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+
+    public function  test_a_logged_in_user_can_create_a_new_post(){
+
+        //create user
+        $user = User::factory()->create();
+
+        //login user
+
+        $response = $this->post('/login', [
```
Comment (ccdilan) [inline tests/Unit/PostTest.php:26]: use actingAs() option to test as logged in user

File: tests/Unit/PostTest.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+       $response = $this->from(route('posts.create'))->post(route('posts.store'),[
+            'title' => 'test title112',
+            'description' => 'test description112'
+        ]);
+
+        // $this->assertEquals(1,Post::count());
+        $response->assertStatus(302); //redirect
+        $response->assertRedirect(route('posts.index'));
+
+
```
Comment (ccdilan) [inline tests/Unit/PostTest.php:43]: need database test also

File: tests/Unit/PostTest.php [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+    public function test_a_logged_in_user_view_posts(){
+
+           //create user
+           $user = User::factory()->create();
+
+           //login user
+           $response = $this->post('/login', [
```
Comment (ccdilan) [inline tests/Unit/PostTest.php:59]: use actingAs() option to test as logged in user

File: tests/Unit/PostTest.php [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+               'email' =>$user->email,
+               'password' => 'password'
+           ]);
+           $this->assertAuthenticated();
+
+           //create a post
+          $response = $this->from(route('posts.create'))->post(route('posts.store'),[
+               'title' => 'test title112',
+               'description' => 'test description112'
+           ]);
```
Comment (ccdilan) [inline tests/Unit/PostTest.php:69]: there are no related assertions for this test case.

File: tests/Unit/PostTest.php [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    public function test_a_logged_in_user_can_update_post(){
+
+        //create user
+        $user = User::factory()->create();
+
+        //login user
+        $response = $this->post('/login', [
```
Comment (ccdilan) [inline tests/Unit/PostTest.php:78]: use actingAs() option to test as logged in user

File: tests/Unit/PostTest.php [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+            'email' =>$user->email,
+            'password' => 'password'
+        ]);
+        $this->assertAuthenticated();
+
+        //create a post
+       $response = $this->from(route('posts.create'))->post(route('posts.store'),[
```
Comment (ccdilan) [inline tests/Unit/PostTest.php:85]: user PostFactory to create posts

**Pull request merged**